lib/cfn-model/parser/lambda_function_parser.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'cfn-model/model/lambda_function'
+
+# Compute the role_id for parsed Lambda function
+class LambdaFunctionParser
+  def parse(cfn_model:, resource:)
+    lambda_function = resource
+
+    if lambda_function.role.instance_of?(String)
+      lambda_function.role_id = lambda_function.role
+    else
+      role_value = lambda_function.role.values[0]
```

Comment:
can you explain this code?  is this implicitly a Hash?  at a high level - "role_id" is a synthetic attribute, but what does that represent vs. the underlying "raw" property.

Reply:
After our discussion I modified the lambda function parser logic to find and associate the lambda execution role as an object in the lambda function object.